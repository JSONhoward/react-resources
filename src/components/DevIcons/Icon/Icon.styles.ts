import styled from "styled-components";

export const StyledIcon = styled('div')`
display: flex;
align-items: center;
justify-content: center;
height: 7rem;
width: 7rem;
border-radius: 50%;
overflow: hidden;
box-shadow: 2px 5px 15px rgba(0,0,0,.25);

img {
    height: 105%;
    width: 105%;

}
`