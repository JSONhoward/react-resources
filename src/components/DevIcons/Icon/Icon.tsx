import React from 'react'
import { StyledIcon } from './Icon.styles';

type Props = {
    icon: string,
    title: string,

}

const Icon: React.FC<Props> = ({icon, title}) => {
    return (
        <StyledIcon>
            <img src={icon} alt={title} />
        </StyledIcon>
    )
}

export default Icon
