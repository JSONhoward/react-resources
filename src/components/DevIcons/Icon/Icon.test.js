import React from 'react';
import {create} from 'react-test-renderer'
import {screen, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Icon from './'

describe('Icon Tests', () => {
    test('Snapshot Tests', () => {
        const tree = create(<Icon />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Renders props', () => {
        const testTitle = 'test title',
            testIcon = 'test icon'

        render(<Icon title={testTitle} icon={testIcon} />)

        expect(screen.getByTestId(`devIcon${testTitle}`)).toHaveAttribute('src', testIcon)
        expect(screen.getByTestId(`devIcon${testTitle}`)).toHaveAttribute('alt', testTitle)
    })
})