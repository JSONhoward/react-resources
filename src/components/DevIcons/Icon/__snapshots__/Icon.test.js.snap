// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Tests Snapshot Tests 1`] = `
<div
  className="sc-bdnylx dYCbwK"
>
  <img
    data-testid="devIconundefined"
  />
</div>
`;
